OMCompiler/Compiler/NFFrontEnd/NFFlatten.mo

```diff
@@ -335,6 +335,58 @@ algorithm
   InstUtil.dumpFlatModelDebug("connections", flatModel);
 end flatten;
 
+function flattenConnection
```

Comment:
This is almost entirely a copy of `flatten`, it would be good if we could avoid duplicating code as much as possible.

I think it might be better to just add a flag to `flatten` instead (e.g. `input Boolean resolveConnections = true`) that can be used to conditionally skip resolving the connections, and have this function just call `flatten` instead of doing all of it itself.